File: LeetCode/C++/trees-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void postorderTraversalUtil(TreeNode* root, vector<int> &tree) {
        if (root) {
            postorderTraversalUtil(root->left, tree);
            postorderTraversalUtil(root->right, tree);
            tree.push_back(root->val);
        }
        
        return;        
    }

    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> tree;
        postorderTraversalUtil(root, tree);
        
        return tree;        
    }
};

File: LeetCode/Python/trees-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        tree = list()
        tree = self.postorderTraversalUtil(root, tree)
        return tree
    
    
    def postorderTraversalUtil(self, root: TreeNode, tree: List[int]) -> List[int]:
        if root:
            tree = self.postorderTraversalUtil(root.left, tree)
            tree = self.postorderTraversalUtil(root.right, tree)
            tree.append(root.val)
            
        return tree
